File: php/Vehicle.php
<?php
abstract class Vehicle{
    protected string $vehicleId;
    protected string$model;
     protected float $fuelLevel;

     public function __construct(string $vehicleId, string $model,$fuelLevel){
        $this->vehicleId = $vehicleId;
        $this->model = $model;
        $this->fuelLevel = $fuelLevel;
     }

     public function refuel (float $liters){
        $this->fuelLevel += $liters;
        return $this->fuelLevel;
     }

     abstract public function calculateRange(): string;

















}


























?>

File: C#/Vehicle.cs
using System;

abstract class Vehicle
{
    protected string VehicleId { get; set; }
    protected string Model { get; set; }
    protected float FuelLevel { get; set; }

    protected Vehicle(string vehicleId, string model, float fuelLevel)
    {
        VehicleId = vehicleId;
        Model = model;
        FuelLevel = fuelLevel;
    }

    public float Refuel(float liters)
    {
        FuelLevel += liters;
        return FuelLevel;
    }

    public abstract string CalculateRange();
}
